projects/core/src/site-theme/facade/site-theme.service.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+
+import { select, Store } from '@ngrx/store';
+import { filter, map, mergeMap, take, tap } from 'rxjs/operators';
+
+import { Observable } from 'rxjs';
+
+import { Theme } from '../../model/misc.model';
+import { SiteThemeConfig } from '../config/site-theme-config';
+import { isNotNullable } from '../../util/type-guards';
+import { StateWithSiteTheme } from '../store/state';
+import { SiteThemeSelectors } from '../store/selectors';
+import { SiteThemeActions } from '../store/actions';
+
+@Injectable()
+export class SiteThemeService {
```

Comment:
In general I appreciate the arch design of this feature is consistent with the design of other site-context features.
That said, I have problem to understand how/where in this PR we actually set the CSS class one the root Angular element (or on `<body>`).
Moreover, AFAIK, we already have an analogical service `ThemeService` which is responsible for setting the theme CSS class on the root Angular element, on the page start. However it takes the theme from the `this.config.context?.[THEME_CONTEXT_ID]?.[0]` see https://github.com/SAP/spartacus/blob/17ad050c8e339bca93e9d57b146b2e675fbadbd0/projects/storefrontlib/layout/theme/theme.service.ts#L36

I have problem to understand why/how we're dealing with 2 different Spartacus global configs:
- config.context.theme
- config.siteTheme

Could you help me understand?

Reply:
On the call with Hak, we agreed that
It doesn't make sense the maintain 2 configs that are conflicting a bit. So we will deprecate the old config, which is less capable (it can contain only strings). The new config can contain objects.

As part of the deprecation, we'll
1. deprecate the public const with `THEME_CONTEXT_ID` key for the old config `config.context[THEME_CONTEXT_ID]` 
  ![image](https://github.com/user-attachments/assets/24114486-0a71-41a5-9a2c-9d69e58e79b8)

2. Don't populate this config anymore deriving from the BaseSitesService response (it's a breaking change in behavior so wrapped in a feature toggle):
  ![image](https://github.com/user-attachments/assets/de8c3adb-5f80-41c2-8794-972bd7b2cbdb)

3. Don't use the old config to for setting the CSS class on the document, but use the newly introduced ngrx state instead (it's a breaking change in behavior, so wrapped in a feature toggle). Moreover we'll create a long-pending subscription (lasting for the lifetime of the application, that will observe the ngrx state of the SiteTheme and on every emission it would apply the new CSS class on the document - the new value emitted from the ngrx state.
  ![image](https://github.com/user-attachments/assets/ac0e3608-c88b-4c70-a9c8-c4ce71e9176a)

4. We won't set the CSS class in the SiteThemeSwitcher component directly. But we will just set the active theme in the ngrx state. And implicitly (according to above point 3.) it will cause setting this CSS class on the document:
   ![image](https://github.com/user-attachments/assets/df942ad1-435d-4635-a2a8-57cec45bbc3e)

Reply:
done